unlock-app/src/components/lock/Lock.js

```diff
@@ -51,20 +38,43 @@ export const Lock = ({ lock, account, keys, purchaseKey }) => {
 }
 
 Lock.propTypes = {
-  keys: UnlockPropTypes.keys,
+  accessKey: UnlockPropTypes.key,
   lock: UnlockPropTypes.lock,
-  account: UnlockPropTypes.account,
+  transaction: UnlockPropTypes.transaction,
   purchaseKey: PropTypes.func,
 }
 
 const mapDispatchToProps = dispatch => ({
   purchaseKey: (key) => dispatch(purchaseKey(key)),
 })
 
-const mapStateToProps = state => {
+export const mapStateToProps = (state, {lock}) => {
+  const account = state.network.account
+  const keys = state.keys
+
+  // If there is no account (probably not loaded yet), we do not want to create a key
+  if(!account) {
```

Comment:
World's tiniest nit: there should be a space here, right?

Reply:
you are right!